File: Java/Spark/SparkSample_JAVA/src/main/java/com/spark_java/sample/Action_AggregateByKey.java
package com.spark_java.sample;


import java.util.Arrays;
import java.util.List;
//import java.util.regex.Pattern;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;


public class Action_AggregateByKey {
	
	//private static final Pattern SPACE = Pattern.compile(" ");
	
	public static void main(String[] args) throws Exception {
		
		JavaSparkContext ctx = new JavaSparkContext(new SparkConf().setAppName("Transform_MapPartition")
				.setMaster("local[*]"));
		
		List<String> data = Arrays.asList("tt","rrrr","eee","www","qqq");
		JavaRDD<String> distData = ctx.parallelize(data,2);

		distData.saveAsObjectFile("C:/scala-workspace/edu-test-file");
		ctx.objectFile("C:/scala-workspace/edu-test-file");
		/*rdd.saveAsObjectFile("C:/scala-workspace/edu-test-file")
	    val loadRdd = sc.objectFile("C:/scala-workspace/edu-test-file")*/
		
		ctx.stop();
		ctx.close();				 
	   
	}
	  
}

File: Java/Spark/spark-edu/src/main/scala/com/bistel/edu/ActionAggregateByKey.scala
package com.bistel.edu

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext;
import org.apache.spark.SparkContext._;

import org.apache.spark.TaskContext;

object ActionAggregateByKey extends App {
  
  def printnum(key : String, x :Any){
    
    println(key)
    println("======================")
    println(x)
  }

  val conf = new SparkConf().setAppName("ActionAggregateByKey").setMaster("local[*]")
  val sc = new SparkContext(conf)
  
  
  //val rdd = sc.parallelize(List(1, 2, 3, 4, 5, 6, 7), 2).collect()
  val rdd = sc.parallelize(List(1, 2, 3, 4, 5, 6), 2)
  //rdd1.foreach(println)
  val a =  rdd.map(x =>(x%2,x))
  
  a.foreach(println)
  
  val b = a.aggregateByKey(10)(math.max(_,_), _+_)

  
  println("b======================")
  b.foreach(println)
  
  
}